src/services/codefixes/fixAwaitInSyncFunction.ts

```diff
@@ -9,19 +9,42 @@ namespace ts.codefix {
         errorCodes,
         getCodeActions(context) {
             const { sourceFile, span } = context;
-            const node = getNodeToInsertBefore(sourceFile, span.start);
-            if (!node) return undefined;
-            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, sourceFile, node));
+            const token = getTokenAtPosition(sourceFile, span.start, /*includeJsDocComment*/ false);
+            const containingFunction = getContainingFunction(token);
+            const insertBefore = getNodeToInsertBefore(sourceFile, containingFunction);
+            const returnType = getReturnTypeNode(containingFunction);
+            if (!insertBefore) return undefined;
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, sourceFile, insertBefore, returnType));
             return [{ description: getLocaleSpecificMessage(Diagnostics.Convert_to_async), changes, fixId }];
         },
         fixIds: [fixId],
-        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) =>
-            doChange(changes, context.sourceFile, getNodeToInsertBefore(diag.file, diag.start!))),
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const token = getTokenAtPosition(diag.file, diag.start!, /*includeJsDocComment*/ false);
+            const containingFunction = getContainingFunction(token);
+            const insertBefore = getNodeToInsertBefore(diag.file, containingFunction);
+            const returnType = getReturnTypeNode(containingFunction);
+            if (insertBefore) {
+                doChange(changes, context.sourceFile, insertBefore, returnType);
+            }
+        }),
     });
 
-    function getNodeToInsertBefore(sourceFile: SourceFile, pos: number): Node | undefined {
-        const token = getTokenAtPosition(sourceFile, pos, /*includeJsDocComment*/ false);
-        const containingFunction = getContainingFunction(token);
+    function getReturnTypeNode(containingFunction: FunctionLike): TypeNode | undefined {
+        switch (containingFunction.kind) {
+            case SyntaxKind.MethodDeclaration:
+            case SyntaxKind.FunctionDeclaration:
+                return containingFunction.type;
+            case SyntaxKind.FunctionExpression:
+            case SyntaxKind.ArrowFunction:
+                if (isVariableDeclaration(containingFunction.parent) &&
+                    containingFunction.parent.type &&
+                    isFunctionTypeNode(containingFunction.parent.type)) {
+                    return containingFunction.parent.type.type;

```

Comment:
Also check for a `.type` immediately on either of these, i.e. `function(): number { ... }` or `(): number => { ... }`.

Reply:
After adding this change I refactored `getReturnType` to not use a switch statement at all since with the added condition the two if statements handle all cases.